src/app/pages/admin/components/user-list/user-list.component.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  Component,
+  OnInit,
+  OnDestroy,
+  signal,
+  model,
+  input,
+  inject,
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { RouterLink } from '@angular/router';
+import { Subject, Observable } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+import { LoadingSpinnerComponent } from '../../../../shared/components/loading-spinner/loading-spinner.component';
+import {
+  BiocommonsUserResponse,
+  FilterOption,
+  ApiService,
+  AdminGetUsersApiParams,
+} from '../../../../core/services/api.service';
+
+@Component({
```

Comment:
Can we have a docstring for this component explaining what needs to be implemented when using it (e.g. what the `getUsers` needs to be